internal/provider/models/modelutils/conditional_types.go

```diff
@@ -125,3 +128,89 @@ func GetAttributesByLevel(level int) map[string]schema.Attribute {
 func GetChildExpressionGroupTypeByLevel(level int) attr.Type {
 	return childExpressionGroups[level].Type()
 }
+
+func UnwindConditionalToModel(component map[string]any) ObjectValue {
+	value, _ := parseExpressionsItem(component, 0)
+	return value
+}
+
+func parseExpressionsItem(component map[string]any, level int) (value ObjectValue, isGroup bool) {
+	logicalOperation := "AND" // Default
+	if operation, ok := component["logical_operation"].(string); ok {
+		logicalOperation = operation
+	}
+	if childExpressionArr, ok := component["expressions"].([]any); ok {
+		// Branch
+		groupItems := make([]attr.Value, 0)
+		leafItems := make([]attr.Value, 0)
+		attributeTypes := ToAttrTypes(GetAttributesByLevel(level))
+
+		for _, e := range childExpressionArr {
+			child := e.(map[string]any)
+			value, isGroup := parseExpressionsItem(child, level+1)
+
+			if isGroup {
+				groupItems = append(groupItems, value)
+			} else {
+				leafItems = append(leafItems, value)
+			}
+		}
+
+		attributeValues := map[string]attr.Value{
+			"logical_operation": NewStringValue(logicalOperation),
+			"expressions":       NewListNull(NestedExpressionAttribute.Type()), // Default to match `plan` since there might only be group expressions on this level
+		}
+
+		if len(leafItems) > 0 {
+			attributeValues["expressions"] = NewListValueMust(NestedExpressionAttribute.Type(), leafItems)
+		}
+
+		expressionGroupType := GetChildExpressionGroupTypeByLevel(level)
+		if len(groupItems) > 0 {
+			attributeValues["expressions_group"] = NewListValueMust(expressionGroupType, groupItems)
+		} else if attributeTypes["expressions_group"] != nil {
+			attributeValues["expressions_group"] = NewListNull(expressionGroupType)
+		}
+
+		rootHasExpressionsAndGroups := len(leafItems) > 0 && len(groupItems) > 0
+		if !rootHasExpressionsAndGroups {
+			return NewObjectValueMust(attributeTypes, attributeValues), true
+		}
+
+		// prepend leaf item expressions to the expressions group
```

Comment:
IIUC the logic should be: "When there's a mix of expressions and expression groups, convert the expressions to expression groups"

Reading the code, that logic is not very clear and there's repeated code.

Can we detect early if there are both `expressions` and `expression_groups` as child, convert them in place and add them to the groups items?